File: src/dev/danzou/kotlin/Day1.kt
package dev.danzou.kotlin

import dev.danzou.kotlin.utils.AdventTestRunner

internal class Day1 : AdventTestRunner() {
    override fun part1(input: String): Number =
        input.split("\n\n")
            .map { it.split("\n").map { it.toLong() } }
            .map { it.sum() }
            .max()

    override fun part2(input: String): Number =
        input.split("\n\n")
            .map { it.split("\n").map { it.toLong() } }
            .map { it.sum() }
            .sorted()
            .takeLast(3)
            .sum()
}

File: src/main/dev/danzou/advent22/scala/Day1.scala
package dev.danzou.advent22.scala

import dev.danzou.advent.utils.UtilsKt._

object Day1 {
  def main(args: Array[String]): Unit = {
    val input = readInputString()

    val sum = input
      .split("\n\n")
      .map({ _.split("\n").map({ _.toLong }) })
      .map({ _.sum })
      .sorted
      .takeRight(3)
      .sum

    println(sum)

    assert(209481 == sum)
  }
}